yt_dlp/extractor/bluesky.py

```diff
@@ -0,0 +1,223 @@
+            'webpage_url': 'https://bsky.app/profile/de1.pds.tentacle.expert/post/3l3w4tnezek2e',
+            'tags': 'count:1',
+            'subtitles': 'count:0',
+            'comments': None,
+        },
+    }]
+
+    def _get_comments(self, meta):
+        yield from self.traverse_replies(meta, traverse_obj(meta, ('post', 'uri'), default=''))
+
+    def traverse_replies(self, thread_node, root_uri):
+        post_uri = traverse_obj(thread_node, ('post', 'uri'))
+        if post_uri != root_uri:
+            post = thread_node.get('post')
+            parent_uri = traverse_obj(post, ('record', 'reply', 'parent', 'uri'))
+            author_handle = traverse_obj(post, ('author', 'handle'))
+            author_did = traverse_obj(post, ('author', 'did'), default='')
+            yield {
+                'id': post_uri,
+                'text': traverse_obj(post, ('record', 'text')),
+                'timestamp': parse_iso8601(traverse_obj(post, ('record', 'createdAt'))),
+                'parent': 'root' if parent_uri == root_uri else parent_uri,
+                'like_count': post.get('likeCount'),
+                'author': traverse_obj(post, ('author', 'displayName')),
+                'author_id': author_did,
+                'author_thumbnail': traverse_obj(post, ('author', 'avatar'), expected_type=url_or_none),
+                'author_url': f'https://bsky.app/profile/{author_handle}',
+                'author_is_uploader': author_did in root_uri,
+            }
+        if replies := thread_node.get('replies'):
+            for reply in replies:
+                yield from self.traverse_replies(reply, root_uri)
+        if parent := thread_node.get('parent'):
+            yield from self.traverse_replies(parent, root_uri)
+
+    def _real_extract(self, url):
+        handle, video_id = self._match_valid_url(url).groups()
+        did = handle if handle.startswith('did:') else self._download_json(
+            'https://public.api.bsky.app/xrpc/com.atproto.identity.resolveHandle',
+            video_id, query={'handle': handle}).get('did')
+
+        getcomments = self.get_param('getcomments', False)
+        meta = self._download_json(
+            'https://public.api.bsky.app/xrpc/app.bsky.feed.getPostThread',
+            video_id, headers={'Content-Type': 'application/json'}, query={
+                'uri': f'at://{did}/app.bsky.feed.post/{video_id}',
+                'depth': 1000 if getcomments else 0,
+                'parentHeight': 1000 if getcomments else 0,
+            }).get('thread')
+        post, record_embed = meta.get('post'), traverse_obj(meta, ('post', 'record', 'embed'))
+
+        formats, subs = self._extract_m3u8_formats_and_subtitles(
+            traverse_obj(post, ('embed', 'playlist'), ('embed', 'media', 'playlist')),
+            video_id, 'mp4', 'm3u8_native', m3u8_id='hls', fatal=False,
+            note='Downloading HD m3u8 information', errnote='Unable to download HD m3u8 information')
```

Comment:
You should differentiate between the different kinds of post response. This will break for posts with embeds such as `https://bsky.app/profile/yunayuispink.bsky.social/post/3l7gqcfes742o`
You also don't need to define the complete path always.
```py
traverse_obj(post, ('embed', 'playlist'), ('embed', 'media', 'playlist'))
```
can be simplified to
```py
traverse_obj(post, ('embed', ('media', None), 'playlist'))
```